pkg/kubernetes/upgrade.go

```diff
@@ -0,0 +1,137 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package kubernetes
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/dapr/cli/pkg/print"
+	"github.com/dapr/cli/utils"
+	helm "helm.sh/helm/v3/pkg/action"
+	"k8s.io/helm/pkg/strvals"
+)
+
+var crds = []string{
+	"components",
+	"configuration",
+	"subscription",
+}
+
+type UpgradeConfig struct {
+	RuntimeVersion string
+}
+
+func Upgrade(conf UpgradeConfig) error {
+	sc, err := NewStatusClient()
+	if err != nil {
+		return err
+	}
+
+	status, err := sc.Status()
+	if err != nil {
+		return err
+	}
+
+	if len(status) == 0 {
+		return errors.New("dapr is not installed in your cluster")
+	}
+
+	print.InfoStatusEvent(os.Stdout, "Dapr control plane version %s detected", status[0].Version)
+
+	helmConf, err := helmConfig(status[0].Namespace)
+	if err != nil {
+		return err
+	}
+
+	daprChart, err := daprChart(conf.RuntimeVersion, helmConf)
+	if err != nil {
+		return err
+	}
+
+	upgradeClient := helm.NewUpgrade(helmConf)
+	upgradeClient.ResetValues = true
+	upgradeClient.Namespace = status[0].Namespace
+	upgradeClient.CleanupOnFail = true
+	upgradeClient.Wait = true
+
+	print.InfoStatusEvent(os.Stdout, "Starting upgrade...")
+
+	mtls, err := IsMTLSEnabled()
+	if err != nil {
+		return err
+	}
+
+	var vals map[string]interface{}
+
+	if mtls {
+		secret, sErr := getTrustChainSecret()
+		if sErr != nil {
+			return sErr
+		}
+
+		ca := secret.Data["ca.crt"]
+		issuerCert := secret.Data["issuer.crt"]
+		issuerKey := secret.Data["issuer.key"]
+
+		ha := highAvailabilityEnabled(status)
+		vals, err = mtlsChartValues(string(ca), string(issuerCert), string(issuerKey), ha)
+		if err != nil {
+			return err
+		}
+	}
+
+	err = applyCRDs(fmt.Sprintf("v%s", conf.RuntimeVersion))
+	if err != nil {
+		return err
+	}
+
+	if _, err = upgradeClient.Run("dapr", daprChart, vals); err != nil {
```

Comment:
make a note somewhere that `dapr` is the release name that it looks to upgrade in helm. 
If it is installed under a different release name in helm and then upgrade is used it won't work. 
Can a `--release-name` flag be provided?

Reply:
I don't want to put Helm specific constructs in the command, also I don't think there's a need here: Helm users will know to list the installation with `helm list`, so specifying the release seems redundant here. If a user installed Dapr through Helm with a non-default release name, they should upgrade using Helm.

Reply:
Can you make a note regarding that in README? 
Also there needs to be addition of upgrade command in the rc3 docs.

Reply:
This is now changed, in last commit I changed the behavior and now the upgrade command will work with any Helm installation of Dapr.